core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/HDTParser.java

```diff
@@ -0,0 +1,311 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.rio.hdt;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.io.input.CountingInputStream;
+
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.RDFHandlerException;
+import org.eclipse.rdf4j.rio.RDFParseException;
+import org.eclipse.rdf4j.rio.RioSetting;
+
+import org.eclipse.rdf4j.rio.hdt.part.HDTDictionary;
+import org.eclipse.rdf4j.rio.hdt.part.HDTDictionarySection;
+import org.eclipse.rdf4j.rio.hdt.part.HDTDictionarySectionFactory;
+import org.eclipse.rdf4j.rio.hdt.part.HDTGlobal;
+import org.eclipse.rdf4j.rio.hdt.part.HDTHeader;
+import org.eclipse.rdf4j.rio.hdt.part.HDTTriples;
+import org.eclipse.rdf4j.rio.hdt.part.HDTTriplesSection;
+import org.eclipse.rdf4j.rio.hdt.part.HDTTriplesSectionFactory;
+
+import org.eclipse.rdf4j.rio.helpers.AbstractRDFParser;
+
+/**
+ * RDF parser for HDT v1.0 files. This parser is not thread-safe, therefore its public methods are synchronized.
```

Comment:
It looks like the thread-safety issues (three publicised fields) are not necessary currently and this could infact be a thread-safe class with minimal changes.